src/Common/Misc/AnnotationBuilder/index.js

```diff
@@ -67,6 +67,28 @@ function fork(annotationObj) {
   return Object.assign({}, annotationObj, { generation, id });
 }
 
+function setDefaultName(annotationObject) {
+  if (annotationObject.selection.type === 'range') {
+    const rangeNames = Object.keys(annotationObject.selection.range.variables);
+    if (rangeNames.length > 0) {
+      annotationObject.name = rangeNames[0];
+      if (rangeNames.length > 1) {
+        annotationObject.name += ` & ${rangeNames[1]}`;
+      }
+      if (rangeNames.length > 2) {
+        annotationObject.name += ' &...';
+      }
+    } else {
+      annotationObject.name = 'empty';
```

Comment:
Is this an actual possibility?  I would have thought we couldn't get a range annotation with no variables in the selection.